File: 6.迭代器模式/geqiangbao_20201125_iterator_swift/geqiangbao_20201125_iterator_swift/demo/AnyIterator.swift
//
//  AnyIterator.swift
//  geqiangbao_20201125_iterator_swift
//
//  Created by xiaobao on 2020/11/25.
//

import UIKit

class AnyIterator<K,M>: NSObject,IteratorProtocol {
    typealias T = M
    var index:Int! = 0
    var datas:K?

    init(datas:K) {
        self.datas = datas
    }
    
    func next() -> M? {
        return nil
    }
    
    func hasNext() -> Bool {
        return false
    }
    


    
  
    
    

}

File: java设计模式/src/迭代器模式/AnyIterator.java
package 迭代器模式;

public class AnyIterator<K,M> implements Iterator{
    public int index;
    public  K datas;

    public AnyIterator(K datas) {
        this.datas = datas;
    }

    @Override
    public M next() {
        return null;
    }

    @Override
    public Boolean hasNext() {
        return null;
    }
}
